Add LDAP discovery via DNS to docs (#56309)
gravitational/teleport: docs/pages/enroll-resources/desktop-access/active-directory.mdx

   windows_desktop_service:
     enabled: true
     ldap:

>        # Address of single LDAP server.
>        # Port must be included for the addr.
>        # LDAPS port is 636 by default (example.com:636)
>        addr: "$LDAP_SERVER_ADDRESS"

Reviewer: ```suggestion
       # Address (including port) of single LDAP server.
       # Can be omitted if `locate_server` is enabled.
       # LDAPS port is 636 by default (example.com:636)
       addr: "$LDAP_SERVER_ADDRESS" 
```

   windows_desktop_service:
     enabled: true
     ldap:
       # Address (including port) of an LDAP server.
       # Can be omitted if `locate_server` is enabled.
       # LDAPS port is 636 by default (example.com:636)

>        addr: "$LDAP_SERVER_ADDRESS" 

Reviewer: Personally, I'd omit `locate_server` from this section. We're not trying to cover every possible configuration option here (this is just step 1), but instead trying to give the reader a starting point.

   address of an LDAP server. There are two ways to specify the LDAP server, either
   by providing a single LDAP server address in the `addr` field, or by enabling
   `locate_server` to automatically discover the LDAP server from DNS SRV records.

   If you opt to use `addr`, for the best results, the address should point to a
   highly-available endpoint to avoid downtime. The port for LDAPS should also be
   included, which is typically 636.

   <details>
   <summary>Example of using the addr field</summary>
   ```yaml
   windows_desktop_service:
      enabled: true
      ldap:
         addr: ldap.example.com:636
         domain: example.com
   ```
   </details>

   Alternatively, you can enable `locate_server` to automatically discover an
   LDAP server from the DNS SRV records of your specified `domain`. This is useful
   if you have multiple LDAP servers for high availability and load balancing.
   Optionally, you can specify a `site` to limit the discovery to a specific
   Active Directory site.

   <details>
   <summary>Example of using locate_server</summary>
   ```yaml
   windows_desktop_service:
      enabled: true
      ldap:
         locate_server: 

>             enable: true

Reviewer: Is it `enable` or `enabled`? This PR switches back and forth between the two.

Author: Should be enabled, I've fixed it now

   For a detailed description of the configuration fields, see
   [Desktop Configuration Reference](../../reference/agent-services/desktop-access-reference/configuration.mdx).

   More information on the differences between the `addr` and `locate_server` options

>    is available in the [LDAP server selection](#ldap-server-selection) section.

Reviewer 2: We generally want to avoid adding links within the numbered instructions so the user doesn't need to skip around the guide and experience unnecessary cognitive load. Is there a way to work these instructions into the numbered section? 

If this is difficult, no worries, but I would at least indicate in this paragraph what kind of information the user would expect to find in the LDAP server selection section so they can decide whether they need to navigate to that section.

Author: Thanks @ptgott, that looks good

In Active Directory environments, Teleport can be configured to discover hosts
via LDAP. LDAP discovery is enabled by adding one or more discovery configs to
the `discovery_configs` field in the Teleport Windows Desktop Service

> configuration. You can set `base_dn` to a wildcard `'*'` to search from the root

Reviewer: Good catch - let's open PRs in the release branches to fix this too.

- manually set the `teleport.dev/computer_name` label to the proper hostname
- ensure that the host's `addr` is its actual hostname

> ## LDAP Server Selection

Author: These settings are mentioned under step 6/7, bullet point 5, but perhaps I should link to this heading for further reading when users are going through the steps
